cli/src/commands/extensions/permissions.rs

```diff
@@ -230,36 +226,39 @@ pub fn default_sandbox() -> SandboxResult<Birdcage> {
 /// Add an execption to the sandbox, ignoring invalid path errors.
 #[cfg(unix)]
 pub fn add_exception(birdcage: &mut Birdcage, exception: Exception) -> SandboxResult<()> {
-    match birdcage.add_exception(exception) {
-        Ok(_) => Ok(()),
+    birdcage.add_exception(exception).map(|_| ()).or_else(|e| match e {
         // Ignore invalid path errors.
-        Err(SandboxError::InvalidPath(_)) => Ok(()),
-        Err(err) => Err(err),
-    }
+        SandboxError::InvalidPath(_) => Ok(()),
+        err => Err(err),
+    })
 }
 
 /// Resolve non-absolute bin paths from `$PATH`.
-pub fn resolve_bin_path(bin: &str) -> PathBuf {
+pub fn resolve_bin_path(bin: &str) -> Result<PathBuf> {
```

Comment:
Having to change this to a `Result` here is another good argument for not resolving home paths inside this function.

Could even make it accept something like `AsRef<Path>` or something to make it absolutely clear that this function only takes an already valid path as argument.

Reply:
Yeah, I'd honestly prefer this to _not_ return `Result<_>`.